hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -247,6 +266,32 @@ private Long delayOffsetCalculation(Option<String> lastCheckpointStr, Set<TopicP
     return delayCount;
   }
 
+  /**
+   * Get the checkpoint by timestamp.
+   * @param consumer
+   * @param topicName
+   * @param timestamp
+   * @return
+   */
+  private String getOffsetsByTimestamp(KafkaConsumer consumer, List<PartitionInfo> partitionInfoList, String topicName, Long timestamp) {
```

Comment:
Can we add tests for the new code that is added. I don't see any tests.

Reply:
When the implementation plan is confirmed, I will quickly add test
When the program is confirmed, I will quickly add test

Reply:
When the program is confirmed, I will quickly add test